doc/rst/language/spec/modules.rst

```diff
@@ -292,17 +309,106 @@ it contains are accessible via the use statement (:ref:`Using_Modules`), import
 naming (:ref:`Explicit_Naming`).
 
+.. _Using_And_Importing:
+
+Using and Importing
++++++++++++++++++++
+
+The ``use`` statement and the ``import`` statement are the two primary ways to
+access a module's symbols from outside of the module.  For top-level modules, a
+``use`` or ``import`` statement is required before referring to the module’s
+name or the symbols it contains within a given lexical scope.
+
+The names that are made visible by a ``use`` or ``import`` statement are
+inserted in to a new scope that immediately encloses the scope within which the
+statement appears.  This implies that the position of the ``use`` or ``import``
+statement within a scope has no effect on its behavior.  If a scope includes
+multiple ``use`` statements, multiple ``import`` statements, or a combination of
+``import`` and ``use`` statements, then the newly-visible names are inserted
+into a common enclosing scope.
+
+.. _Use_And_Import_Conflicts:
+
+Conflicts
++++++++++
+
+The scope inserted by the presence of ``use`` and ``import`` statements follows
+the same rules about conflicting symbols as symbols defined in the same scope
+(see :ref:`Variable_Conflicts`).  Thus an error will be signaled if multiple
+conflicting public module-level symbols would be inserted into this enclosing
+scope with the same name, and that name is accessed by other statements in the
+same scope as the ``use`` or ``import``.  Remember that this does not apply to
+functions unless they are also indistinguishable in other ways, see
+:ref:`Function_Overloading`.
+
+Because symbols brought into scope by a ``use`` or ``import`` statement are
+placed at a scope enclosing where the statement appears, such symbols will be
+shadowed by other symbols with the same name defined in the scope with the
+statement.  Thus the symbols that are shadowed will only be accessible via
+:ref:`Explicit_Naming`.
+
+Symbols defined by public ``use`` or ``import`` statements can impact the scope
+they are inserted into in different ways (see :ref:`Public_Use` and
+:ref:`Reexporting` for more information on the ``public`` keyword).  Symbols
+that are brought in by a ``public use`` for unqualified access are treated as
+at successive distances relative to how many ``public use`` statements were
+necessary to obtain them.  For instance,
+
+   *Example (conflict1.chpl)*.
+
+   .. code-block:: chapel
+
+      module A {
+        var x: int;
+      }
+
+      module B {
+        public use A;
+      }
+
+      module C {
+        var x: bool;
+      }
+
+      module MainMod {
+        use B, C;
+
+        proc main() {
+          writeln(x);
+        }
+      }
+
+   This code demonstrates a module (MainMod) using two modules, B and C.  Module
+   C defines a symbol named x, while module B publicly uses another module, A,
+   which also defines a symbol named x.  The program as written will not
+   conflict and will print out the value of ``C.x``, which is ``false``, because
```

Comment:
"will not conflict" does not seem to be a specified term. How about "will compile" ?